File: frontend/services/inventory.service.ts
import { apiClient } from "@/lib/axios";
import type { InventoryItem, InventoryUpdatePayload } from "@/types";

export const inventoryService = {
  list: async (): Promise<InventoryItem[]> => {
    const { data } = await apiClient.get<InventoryItem[]>("/api/inventory");
    return data;
  },

  lowStock: async (): Promise<InventoryItem[]> => {
    const { data } = await apiClient.get<InventoryItem[]>("/api/inventory/low-stock");
    return data;
  },

  addBatch: async (medicationId: number, payload: InventoryUpdatePayload): Promise<InventoryItem> => {
    const { data } = await apiClient.post<InventoryItem>(`/api/inventory/${medicationId}`, payload);
    return data;
  },

  update: async (itemId: number, payload: InventoryUpdatePayload): Promise<InventoryItem> => {
    const { data } = await apiClient.patch<InventoryItem>(`/api/inventory/${itemId}`, payload);
    return data;
  },
};

File: mobile/lib/data/services/inventory_service.dart
import 'package:dio/dio.dart';
import '../../core/constants/api_endpoints.dart';
import '../models/inventory_model.dart';

class InventoryService {
  InventoryService(this._dio);
  final Dio _dio;

  Future<List<InventoryItemModel>> list() async {
    final response = await _dio.get(ApiPaths.inventory);
    return (response.data as List<dynamic>).map((e) => InventoryItemModel.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<List<InventoryItemModel>> lowStock() async {
    final response = await _dio.get(ApiPaths.inventoryLowStock);
    return (response.data as List<dynamic>).map((e) => InventoryItemModel.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<InventoryItemModel> addBatch(int medicationId, Map<String, dynamic> payload) async {
    final response = await _dio.post(ApiPaths.inventoryAddBatch(medicationId), data: payload);
    return InventoryItemModel.fromJson(response.data as Map<String, dynamic>);
  }

  Future<InventoryItemModel> update(int itemId, Map<String, dynamic> payload) async {
    final response = await _dio.patch(ApiPaths.inventoryUpdate(itemId), data: payload);
    return InventoryItemModel.fromJson(response.data as Map<String, dynamic>);
  }
}
